SqlInstall.ps1

```diff
@@ -0,0 +1,556 @@
+        Write-Host 'Setup will start from ' + $SetupFile
+    } 
+    else {
+        $SetupFile = $SetupFile + '\SQLEXPR_x64_ENU\SETUP.EXE'
+        Write-Host 'Setup will start from ' + $SetupFile
+    }
+
+
+    $ConfigFile = 'c:\temp\'
+
+    if ( -Not (Test-Path -Path $ConfigFile ) ) {
+        New-Item -ItemType directory -Path $ConfigFile
+    }
+
+    Out-File -FilePath C:\Temp\ConfigurationFile2.ini -InputObject $startScript
+
+    $FileLocation2 = $ConfigFile + 'ConfigurationFile2.ini'
+
+    (Get-Content -Path $FileLocation2).Replace('SQLBACKUPDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Backup"', 'SQLBACKUPDIR="' + $BackupVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Backup"') | Out-File $FileLocation2
+
+    (Get-Content -Path $FileLocation2).Replace('SQLUSERDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLUSERDBDIR="' + $DataVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $FileLocation2
+
+    (Get-Content -Path $FileLocation2).Replace('SQLTEMPDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLTEMPDBDIR="' + $TempVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $FileLocation2
+
+    (Get-Content -Path $FileLocation2).Replace('SQLUSERDBLOGDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"', 'SQLUSERDBLOGDIR="' + $LogVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"') | Out-File $FileLocation2
+
+    (Get-Content -Path $FileLocation2).Replace('SQLSYSADMINACCOUNTS="WIN-NAJQHOBU8QD\Administrator"', 'SQLSYSADMINACCOUNTS="' + $env:COMPUTERNAME + '\Administrator"')| Out-File $FileLocation2
+
+    #$SetupFile = 'C:\Users\Administrator\Downloads\SQLEXPR_x64_ENU\Setup.exe'
+    #$ConfigFile = 'C:\temp\ConfigurationFile2.ini'
+
+    $SAPassW = '[InsertPasswordHere]'
+
+    & $SetupFile /ConfigurationFile=$FileLocation2 /Q /IACCEPTSQLSERVERLICENSETERMS /SAPWD=$SAPassW
+
+    # Grant service account the right to perform volume maintenance
+    # code found at https://social.technet.microsoft.com/Forums/windows/en-US/5f293595-772e-4d0c-88af-f54e55814223/adding-domain-account-to-the-local-policy-user-rights-assignment-perform-volume-maintenance?forum=winserverpowershell
+
+    ## <--- Configure here
+    $accountToAdd = 'NT Service\MSSQL$AXIANSDB01'
+    ## ---> End of Config
+    $sidstr = $null
+
+
+    try {
+        $ntprincipal = new-object System.Security.Principal.NTAccount "$accountToAdd"
+        $sid = $ntprincipal.Translate([System.Security.Principal.SecurityIdentifier])
+        $sidstr = $sid.Value.ToString()
+    }
+    catch {
+        $sidstr = $null
+    }
+    Write-Host "Account: $($accountToAdd)" -ForegroundColor DarkCyan
+    if ( [string]::IsNullOrEmpty($sidstr) ) {
+        Write-Host "Account not found!" -ForegroundColor Red
```

Comment:
I'd do validation earlier, and exit out of the command. To exit, we use our internal function 

```powershell
Stop-Function -Message "$accountToAdd could not be found"
return
```